### ISSUE TITLE
spyder crashes with packages ipywidgets or display

### ISSUE THREAD

**Issue by WiMiME** [spyder-ide/spyder#21151]
### What steps will reproduce the problem?

import pandas as pd
import matplotlib.pyplot as plt
import ipywidgets as widgets
from ipywidgets import interact
from IPython.display import display

# Create the pandas DataFrame 'df'
data = {
    'A': [10, 20, 30, 40, 50],
    'B': [5, 10, 15, 20, 25],
    'C': [2, 4, 6, 8, 10]
}
df = pd.DataFrame(data)

# Define the function for the calculation and plot
def update_plot(R):
    df['result'] = df['A'] - df['B'] / df['C'] + R
    plt.figure(figsize=(8, 6))
    plt.plot(df.index, df['result'], marker='o')
    plt.xlabel('Index')
    plt.ylabel('Result')
    plt.title(f"Plot for R = {R}")
    plt.grid(True)
    plt.show()

# Create the interactive slider for R
R_slider = widgets.FloatSlider(value=1.0, min=0.0, max=5.0, step=0.1, description='R:')

# Create the interactive plot using the 'interact' function
interactive_plot = interact(update_plot, R=R_slider)

# Display the interactive plot
display(interactive_plot)

Win10/Python3.8

### Traceback
```python-traceback
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 134, in comm_open
    self.log.error("No such comm target registered: %s", target_name)
Message: 'No such comm target registered: %s'
Arguments: ('jupyter.widget',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 113, in get_comm
    return self.comms[comm_id]
KeyError: '1fcfffa865ac4665bdc301f274c0eb97'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 172, in comm_msg
    comm = self.get_comm(comm_id)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 117, in get_comm
    self.log.warning("No such comm: %s", comm_id)
Message: 'No such comm: %s'
Arguments: ('1fcfffa865ac4665bdc301f274c0eb97',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 113, in get_comm
    return self.comms[comm_id]
KeyError: 'a890ec9d59d7406c88d2077dc9cee261'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 172, in comm_msg
    comm = self.get_comm(comm_id)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 117, in get_comm
    self.log.warning("No such comm: %s", comm_id)
Message: 'No such comm: %s'
Arguments: ('a890ec9d59d7406c88d2077dc9cee261',)
--- Logging error ---
Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 113, in get_comm
    return self.comms[comm_id]
KeyError: 'a890ec9d59d7406c88d2077dc9cee261'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\mme\appdata\local\programs\python\python38\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mme\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\MME\AppData\Local\Programs\Python\Python38\Scripts\spyder.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\start.py", line 213, in main
    mainwindow.main(options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 48, in _dispatch
    getattr(self, msg_type)(msg)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 172, in comm_msg
    comm = self.get_comm(comm_id)
  File "c:\users\mme\appdata\local\programs\python\python38\lib\site-packages\qtconsole\comms.py", line 117, in get_comm
    self.log.warning("No such comm: %s", comm_id)
Message: 'No such comm: %s'
Arguments: ('a890ec9d59d7406c88d2077dc9cee261',)
```

## Versions

* Spyder version: 5.0.0 
* Python version: 3.8.10
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.31.0 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  4.3.3 (OK)
keyring >=17.0.0              :  23.5.0 (OK)
nbconvert >=4.0               :  6.4.5 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.9.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.15.1 (OK)
pylint >=1.0                  :  2.12.2 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.7 (OK)
pyls_spyder >=0.3.2           :  0.4.0 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=0.5.7             :  1.1.1 (OK)
qtconsole >=5.0.3             :  5.2.2 (OK)
qtpy >=1.5.0                  :  2.0.0 (OK)
setuptools >=39.0.0           :  56.0.0 (OK)
sphinx >=0.6.6                :  4.3.2 (OK)
spyder_kernels >=2.0.1,<2.1.0 :  2.0.5 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  3.5.1 (OK)
numpy >=1.7                   :  1.22.0 (OK)
pandas >=1.1.1                :  1.4.3 (OK)
scipy >=0.17.0                :  1.7.3 (OK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @WiMiME, thanks for reporting. This problem is fixed in our latest version available in Anaconda (5.4.3). Please update by closing Spyder, opening the Anaconda Prompt and running there the commands mentioned in [our documentation](http://docs.spyder-ide.org/current/installation.html#updating-with-anaconda).

If that doesn't work, I'm afraid you'll have to remove and reinstall Anaconda.

**Comment by WiMiME** [comment]
Thank you for your quick answer and in this case sry for the inconvenience
